packit/schema_m.py

```diff
@@ -0,0 +1,196 @@
+        **kwargs,
+    ):
+        """
+
+        :param value:
+        :param attr:
+        :param data:
+        :param kwargs:
+        :return:
+
+        :todo: add error handling for value type and content if is dict
+        """
+        file_to_sync = None
+
+        if isinstance(value, dict):
+            file_to_sync = SyncFilesItem(src=value["src"], dest=value["dest"])
+        elif isinstance(value, str):
+            file_to_sync = SyncFilesItem(src=value, dest=value)
+        else:
+            # todo: move to validation method
+            raise ValidationError
+
+        return file_to_sync
+
+
+class SyncFilesConfigSchema(Schema):
+    files_to_sync = fields.List(FilesToSyncField(allow_none=False), allow_none=True)
+
+    @post_load
+    def make_instance(self, data, **kwargs):
+        from packit.config import SyncFilesConfig
+
+        return SyncFilesConfig(**data)
+
+    @pre_load
+    def list_to_dict(self, data, **kwargs):
+        """
+        files are provided as list not as schema dict, input data format is modified
+        to follow schema layout
+
+        ex.
+        [f1,f2,..] -> {"files_to_sync": [f1, f2, f3, ...]}
+        """
+
+        if isinstance(data, list):
+            return {"files_to_sync": data}
+        else:
+            return data
+
+
+class JobConfigSchema(Schema):
+    class Meta:
+        unknown = RAISE
+
+    job = EnumField(JobType, required=True)
+    trigger = EnumField(JobTriggerType, required=True)
+    metadata = fields.Dict(missing={})
+
+    @post_load
+    def make_instance(self, data, **kwargs):
+        return JobConfig(**data)
+
+
+class PackageConfigSchema(Schema):
+    class Meta:
+        unknown = RAISE
+
+    config_file_path = fields.String()
+    specfile_path = fields.String(required=True)
+    downstream_package_name = fields.String()
+    upstream_project_url = fields.String(missing=None)
+    upstream_package_name = fields.String()
+    upstream_ref = fields.String()
+    upstream_tag_template = fields.String()
+    dist_git_url = fields.Method(deserialize="not_processed")
+    dist_git_base_url = fields.String()
+    dist_git_namespace = fields.String()
+    create_tarball_command = fields.List(fields.String())
+    current_version_command = fields.List(fields.String())
+    allowed_gpg_keys = fields.List(fields.String())
+    spec_source_id = fields.Method(deserialize="spec_source_id_method")
+    synced_files = fields.Nested(SyncFilesConfigSchema)
+    jobs = fields.Nested(JobConfigSchema, many=True)
+    actions = fields.Dict(
+        keys=EnumField(ActionName, by_value=True)
+    )  # add value validatation using custom field or validator
+    create_pr = fields.Boolean()
+
+    # list of deprecated keys and their replacement (new,old)
+    deprecated_keys = (
+        ("upstream_package_name", "upstream_project_name"),
+        ("upstream_package_name", "upstream_name"),
+        ("downstream_package_name", "package_name"),
```

Comment:
I'd say that at this point, no one has the old deprecated keys upstream_name and package_name so I'd drop those probably

Reply:
removed